age-calc/app/src/DateChecker.tsx

```diff
@@ -0,0 +1,182 @@
+import DateCard from "./DateCard";
+import "./DateChecker.css";
+import { useState } from "react";
+
+interface DateCheckerProps {
+  year: number | null;
+  month: number | null;
+  day: number | null;
+  setResultYear: (value: number | string) => void;
+  setResultMonth: (value: number | string) => void;
+  setResultDay: (value: number | string) => void;
+}
+
+export default function DateChecker(props: DateCheckerProps) {
+  //birthDay,birthMonth, birthYear - users birth date, starts as string and changes to number, for errorChecks
+  const [birthDay, setBirthDay] = useState("");
+  const [birthMonth, setBirthMonth] = useState("");
+  const [birthYear, setBirthYear] = useState("");
+  //dayMsg, monthMsg, yearMsg - used to set error messages - passed as prop
+  const [dayMsg, setDayMsg] = useState("");
+  const [monthMsg, setMonthMsg] = useState("");
+  const [yearMsg, setYearMsg] = useState("");
+  //missing and warning are used to modify css in child components - passed as prop
+  const [missing, setMissing] = useState("");
+  const [warning, setWarning] = useState("");
+
+  const handleButtonClick = () => {
+    const currentYear = props.year;
+    const currentMonth = props.month;
+    const currentDay = props.day;
```

Comment:
You can use destructuring to get properties out of props object instead of making so many lines to redefine it.